src/app/modules/parts/components/edit-part/edit-part.component.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { FormGroup, FormBuilder, Validators, FormArray, AbstractControl, FormControl } from '@angular/forms';
+import { ActivatedRoute, Params } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { PartsService } from 'src/app/core/services/parts/parts.service';
+import { PartModel } from 'src/app/core/models/part-model';
+import { CurrentUserService } from 'src/app/core/services/currentUser/current-user.service';
+
+@Component({
+  selector: 'app-edit-part',
+  templateUrl: './edit-part.component.html',
+  styleUrls: ['./edit-part.component.css']
+})
+export class EditPartComponent implements OnInit, OnDestroy {
+  protected editForm: FormGroup;
+  protected isEditMode: boolean = false;
+
+  private selectedPart: PartModel;
+
+  private routeSubscription: Subscription;
+  
+  constructor(private fb: FormBuilder,
+              private router: ActivatedRoute,
+              private partsService: PartsService,
+              private currentuser: CurrentUserService) { }
+
+  ngOnInit() {
+    this.initForm();
+
+    this.routeSubscription = this.router.params
```

Comment:
important! here we have memory leak due to absence of unsubscribe.
Lets discuss unsubscribe logic in call.